### PR TITLE
feat: Added model for learner pathway

### PR DESCRIPTION
**Pull request by irfanuddinahmad**
**Description**: This PR adds the LearnerPathway model.

**JIRA**: [ENT-5054](https://openedx.atlassian.net/browse/ENT-5054)

### CODE REVIEW

File: course_discovery/apps/learner_pathway/tests/factories.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+@pytest.mark.django_db
+class LearnerPathwayFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = LearnerPathway
+
+    name = FuzzyText(prefix='learner-pathway-name-')
+    uuid = factory.Faker('uuid4')
+
+
+@pytest.mark.django_db
```
Comment (saleem-latif) [inline course_discovery/apps/learner_pathway/tests/factories.py]: no need to add the decorator here.

File: course_discovery/apps/learner_pathway/tests/test_models.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import ddt
+import pytest
+from django.test import TestCase
+
+from course_discovery.apps.learner_pathway.tests import factories
+
+
+@ddt.ddt
+@pytest.mark.django_db
```
Comment (saleem-latif) [inline course_discovery/apps/learner_pathway/tests/test_models.py:16]: can you switch the order of the decorators?

```
@pytest.mark.django_db
@ddt.ddt
class LearnerPathwayTests(TestCase):
```

File: course_discovery/apps/learner_pathway/models.py [hunk 3]
```diff
@@ -32,3 +32,42 @@ def get_skills(self) -> [str]:
...
+
+class LearnerPathway(models.Model, metaclass=AbstractModelMeta):
+    """
+    Top level model for learner pathway.
+    """
+    uuid = models.UUIDField(default=uuid4, editable=False, unique=True, verbose_name=_('UUID'))
+    name = models.CharField(max_length=255, null=False, blank=False, help_text=_('Pathway name'))
+
+    @property
+    def time_of_completion(self) -> int:
```
Comment (zamanafzal) [inline course_discovery/apps/learner_pathway/models.py]: I think the return type should be a float as per L49.

**Review by saleem-latif — APPROVED** [review]

**Review by muneebGH — APPROVED** [review]
LETS GOOOOO!!!!!! 🥳

**Pull request merged**